pkg/gitprovider/gitee.go

```diff
@@ -0,0 +1,380 @@
+		})
+	}
+	return response, nil
+}
+
+func (g *GiteeGitProvider) GetBranchByCommit(staticContext *StaticGitContext) (string, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+	branches, _, err := client.RepositoriesApi.GetV5ReposOwnerRepoBranches(ctx, staticContext.Owner, staticContext.Name, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to get branch by commit: %w", err)
+	}
+
+	var branchName string
+	for _, branch := range branches {
+		commitId := branch.Commit.Sha
+		if *staticContext.Sha == commitId {
+			branchName = branch.Name
+			break
+		}
+
+		commits, _, err := client.RepositoriesApi.GetV5ReposOwnerRepoCommits(ctx, staticContext.Owner, staticContext.Name, &gitee.GetV5ReposOwnerRepoCommitsOpts{
+			Sha: optional.NewString(branchName),
+		})
+		if err != nil {
+			continue
+		}
+
+		if len(commits) == 0 {
+			continue
+		}
+
+		for _, commit := range commits {
+			if commit.Sha == *staticContext.Sha {
+				branchName = branch.Name
+				break
+			}
+		}
+		if branchName != "" {
+			break
+		}
+	}
+
+	if branchName == "" {
+		return "", fmt.Errorf("status code: %d branch not found for SHA: %s", http.StatusNotFound, *staticContext.Sha)
+	}
+	return branchName, nil
+}
+
+func (g *GiteeGitProvider) GetLastCommitSha(staticContext *StaticGitContext) (string, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+
+	var sha *string
+
+	if staticContext.Branch != nil {
+		sha = staticContext.Branch
+	}
+
+	if staticContext.Sha != nil {
+		sha = staticContext.Sha
+	}
+	commits, _, err := client.RepositoriesApi.GetV5ReposOwnerRepoCommits(ctx, staticContext.Owner, staticContext.Name, &gitee.GetV5ReposOwnerRepoCommitsOpts{
+		Sha: optional.NewString(*sha),
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(commits) == 0 {
+		return "", nil
+	}
+	return commits[0].Sha, nil
+}
+
+func (g *GiteeGitProvider) GetDefaultBranch(staticContext *StaticGitContext) (*string, error) {
+	client := g.getApiClient()
+	ctx := context.Background()
+	repo, _, err := client.RepositoriesApi.GetV5ReposOwnerRepo(ctx, staticContext.Owner, staticContext.Name, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &repo.DefaultBranch, nil
+}
+
+func (g *GiteeGitProvider) GetUrlFromContext(repoContext *GetRepositoryContext) string {
+	url := strings.TrimSuffix(repoContext.Url, ".git")
+
+	if repoContext.Branch != nil && *repoContext.Branch != "" {
+		if repoContext.Sha != nil && *repoContext.Sha == *repoContext.Branch {
+			//https://gitee.com/<owner>/<repo>/commit/<sha>
+			if repoContext.Path != nil {
+				url += "/blob/" + *repoContext.Sha + "/" + *repoContext.Path
+			} else {
+				url += "/commit/" + *repoContext.Branch
+			}
+		} else {
+			if repoContext.Path != nil {
+				url += "/blob/" + *repoContext.Branch + "/" + *repoContext.Path
+			} else {
+				url += "/tree/" + *repoContext.Branch
+			}
+		}
+	} else if repoContext.Path != nil {
+		url += "/blob/master/" + *repoContext.Path
```

Comment:
`master` is default on gitee? not `main`?

Reply:
Yes. I created a new repo, and the default branch was master bydefault.